spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/LazySpanSubscriber.java

```diff
@@ -29,7 +28,7 @@
  * @author Marcin Grzejszczak
  * @since 2.0.0
  */
-final class LazySpanSubscriber<T> extends AtomicBoolean implements SpanSubscription<T> {
+final class LazySpanSubscriber<T> implements SpanSubscription<T> {
```

Comment:
That was a suggestion by @smaldini and this is how the reactor team does it cc @simonbasle

Reply:
ok.. maybe we can add a comment once we figure out why it helps

Reply:
I think some files use compareAndSet, just this one doesn't so the AtomicBoolean is a distraction. that's why I removed it.

I guess if we are supposed to do some only once callbacks, we should use the atomic methods ya?

Reply:
I don't want to be blocked on this as there's more important things. I replaced with a TODO to explain wtf.

Reply:
as far as I understand, this pattern reduces allocation of extra field in case when subscriber has `boolean` state  
and if you do not use `AtomicBoolean` methods it does not make sense to apply it.

Reply:
considering allocations  
maybe it reasonable to merge `SpanSubscriptionProvider` and `LazySpanSubscriber` so you do not allocate extra object?